src/remove-simple-form-value.js

```diff
@@ -30,5 +33,34 @@ export function removeDatasetForSimpleFormValue(value, options) {
   //This returns the complete chain of triples for the path, if there something missing, new nodes are added.
   const dataset = triplesForPath(options, true);
   const triplesToRemove = dataset.triples.filter((t) => t.object.equals(value));
+
+  // Remove dangling FormDataNode types (subjects with only rdf:type form:FormDataNode)
+  const triplesToRemoveLength = triplesToRemove.length;
+
+  for (let index = 0; index < triplesToRemoveLength; index++) {
+    const triple = triplesToRemove[index];
+
+    if (!triple.subject?.value?.startsWith(URI_TEMPLATE)) return;
```

Comment:
`this will be the only triple left for that subject after removal of the triple in triplesToRemove`

https://github.com/lblod/submission-form-helpers/blob/a2fbc0ce26c7828d0d4abb85511e798621efbec2/src/remove-simple-form-value.js#L51-L53

Reply:
That's not what i meant. the logic will break if template URI changed and a form was saved in the previous template.